packages/project/src/merge/map-uuids.ts

```diff
@@ -39,82 +40,234 @@ export interface MappingResults {
 }
 
 export default (source: Workflow, target: Workflow): MappingResults => {
-  const nodeMapping: Record<string, MappingRule> = {};
   const edgeMapping: Record<string, MappingRule> = {};
-  // const mapping: Record<string, MappingRule> = {};
 
-  // sets all nodes as not existing in target
-  const targetIds: Record<string, string> = {};
-  for (const tstep of target.steps) {
-    nodeMapping[tstep.id] = null;
-    targetIds[tstep.id] = tstep.openfn?.uuid || tstep.id;
+  // Map by id
+  let {
+    mapping: nodeMapping,
+    pool,
+    idMap,
+  } = mapStepsById(source.steps, target.steps);
 
-    // dealing with edges
-    const next =
-      typeof tstep.next === 'string'
-        ? { [tstep.next]: true }
-        : tstep.next || {};
-    for (const [toNode, toValue] of Object.entries(next)) {
-      const edgeId = tstep.id + '-' + toNode;
-      edgeMapping[edgeId] = null;
-      targetIds[edgeId] =
-        typeof toValue === 'object' ? toValue.openfn?.uuid || edgeId : edgeId;
+  for (const source_step of pool.source) {
+    if (!source_step.id) continue; // yh. we'll always have it.
+
+    // these are the candidates for the search
+    let candidates = pool.target.filter((step) => !idMap.has(step.id));
```

Comment:
I think think nodes should already have been removed from the pool, doing away with this step